userdata.tpl

```diff
@@ -2,9 +2,134 @@ MIME-Version: 1.0
 Content-Type: multipart/mixed; boundary="//"
 
 --//
-Content-Type: text/x-shellscript; charset="us-ascii"
-#!/bin/bash
+Content-Type: text/x-shellscript
+Content-Type: charset="us-ascii"
 
 ${before_cluster_joining_userdata}
 
---//--
+cat << 'EOF' > /etc/eks/extra_args.sh
+#!/usr/bin/env bash
```

Comment:
While arguably `#!/usr/bin/env bash` is more portable than `#!/bin/bash` it is less secure and if the AMI you are using does not have `bash` installed as `/bin/bash` then what are the chances that is it also using `/etc/systemd/system/kubelet.service.d/30-kubelet-extra-args.conf`? Not great. So I would leave this as `#!/bin/bash`